src/main/java/org/spongepowered/common/mixin/core/tileentity/MixinTileEntityNote.java

```diff
@@ -54,4 +72,42 @@ public void supplyVanillaManipulators(List<DataManipulator<?, ?>> manipulators)
         manipulators.add(getNoteData());
     }
 
+    @Override
+    @Nullable
+    public InstrumentType getOverriddenInstrument() {
+        return this.overriddenInstrument;
+    }
+
+    @Override
+    public void setOverriddenInstrument(@Nullable InstrumentType instrumentType) {
+        this.overriddenInstrument = instrumentType;
+    }
+
+    @Override
+    public void readFromNbt(NBTTagCompound compound) {
+        super.readFromNbt(compound);
+        if (compound.hasKey(OVERRIDDEN_INSTRUMENT_KEY)) {
+            final String instrument = compound.getString(OVERRIDDEN_INSTRUMENT_KEY);
+            InstrumentTypeRegistryModule.getInstance().getById(instrument)
+                    .ifPresent(instrumentType -> this.overriddenInstrument = instrumentType);
+        }
+    }
+
+    @Override
+    public void writeToNbt(NBTTagCompound compound) {
+        super.writeToNbt(compound);
+        if (this.overriddenInstrument != null) {
+            compound.setString(OVERRIDDEN_INSTRUMENT_KEY, this.overriddenInstrument.getId());
+        }
+    }
+
+    @Redirect(method = "triggerNote(Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;)V",
+            at = @At(value = "INVOKE", target = "Lnet/minecraft/world/World;addBlockEvent("
+                    + "Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/block/Block;II)V"))
+    public void onTriggerNote(World target, BlockPos blockPos, Block block, int instrument, int note) {
```

Comment:
No need to have this public (can be private)

Reply:
Fixed